### PR TITLE
More ioport upgrades -> required_ioport_array

### PR DESCRIPTION
**Pull request by enewton**
Removing sprintf etc.
I've tested what I can but a few were not in a working state to begin with.

### CODE REVIEW

File: src/mame/sharp/mz80_m.cpp [hunk 1]
```diff
@@ -25,15 +25,12 @@ void mz80_state::machine_reset()
 	m_mz80k_keyboard_line = 0;
 }
 
-
 uint8_t mz80_state::mz80k_8255_portb_r()
```
Comment (cuavas) [inline src/mame/sharp/mz80_m.cpp:28]: Please put the double blank line separating lifecycle from I/O back.

**Review by cuavas — COMMENTED** [review]
These are all `required_ioport_array` so if anything uses the wrong format it’ll blow up as soon as the machine starts.

File: src/mame/sharp/mz80_m.cpp (hunk 1 above)
Comment (enewton) [reply src/mame/sharp/mz80_m.cpp]: I think you've done it?

Comment (cuavas) [reply src/mame/sharp/mz80_m.cpp]: Yeah, I’m in a hurry.

**Comment by cuavas** [comment]
You've broken quite a few systems in intv.cpp - they aren't marked not working, and they blow up on validation (without even needing to be started). I thought you said you tested the changes? If you'd started even one system from that source file or run validation, you'd see these errors.

**Comment by rb6502** [comment]
I've fixed the issue.  The root of the problem is that the intv drivers are still "old MESS".  The voice module, ECS, and keyboard component were add-ons and need to be refactored that way at some point.

That said, please do run "mame -validate" on future changes.

**Pull request merged**